### ISSUE TITLE
`UnsavedController` does not immediately mark the form as dirty after an invalid submission

### ISSUE THREAD

**Issue by laymonage** [wagtail/wagtail#12355]
### Issue Summary

After submitting the page editor form with invalid data, the form is not immediately marked as dirty. You need to wait until the [initial delay lapses (10s)](https://github.com/wagtail/wagtail/blob/d526612e4befbb1bb80addb596830b38ab0dabde/client/src/controllers/UnsavedController.ts#L13), make some changes, and _then_ the form will be marked as dirty (and will be kept as dirty even if you undo those changes).

<details>

<summary>Screen recording</summary>

https://github.com/user-attachments/assets/d510d88e-3707-4f00-9ca4-0a29d3785266

</details>

This is a change of behaviour compared to how it was before #11373. The previous behaviour was that the form will immediately be marked as dirty, without you needing to wait for the initial delay and without having to make any changes.

<details>

<summary>Screen recording</summary>

https://github.com/user-attachments/assets/88209b5d-df5e-4318-a5f5-f215cc357d7b

</details>

### Steps to Reproduce

1. Start a new project with `wagtail start myproject` with Wagtail 6.0+
2. Try to edit the home page, empty out the title field and submit so that it gives you a validation error
3. Observe that the form is not marked as dirty.
4. You can immediately leave the page without triggering the browser confirmation dialog.
5. If you choose to wait for 10s, then make some changes (without saving), the form will be marked as dirty. The form will still be dirty even if you undo those changes.

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

I don't know if this was intentional in #11373, but I didn't find any comments about it and I think the original behaviour was deliberate. If you submit an invalid form, it's highly likely that you made some changes (thus making it invalid). These changes are not saved yet, so we should remind the editor and prevent them from accidentally losing their changes by leaving the page.

This is what the `alwaysDirty` (now `forceValue`) was meant to do, but unfortunately the intended functionality got lost in the refactoring.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes on bakerydemo

### Working on this

I already have a fix working, will push that as part of #12139.

**Comment by lb-** [comment]
Thanks @laymonage - I thought the initial 10s was intentional and hence tried to emulate that in the migration, my reading of the code at the time indicated that this was required to ensure that there was time for all JS widgets to load (e.g. comments, rich text).

I thought it was odd that we did that, I actually even explicitly tested that it waited 10 seconds before doing an edit. I remember even testing this 10s delay happened on the original code, I did not write a comment about this sorry as I trusted I understood the comments in the code (big mistake).

It looks like we got confused about what this timing was meant for somehow.

See the original code.

https://github.com/wagtail/wagtail/blob/3ee28ee8b2554232a8854035277d442e46975b0c/client/src/entrypoints/admin/core.js#L194

https://github.com/wagtail/wagtail/blob/3ee28ee8b2554232a8854035277d442e46975b0c/client/src/entrypoints/admin/core.js#L152-L153

Essentially, delay the initial snapshot of the 'form data' for 10 seconds to avoid race conditions.

However, your explanation makes sense so I will assume that the refactor behaviour broke this.
